Composer/packages/types/src/runtime.ts

```diff
@@ -63,7 +63,7 @@ export type RuntimeTemplate = {
   eject?: (project: IBotProject, localDisk?: any, isReplace?: boolean) => Promise<string>;
 
   /** build method used for local publish */
-  build: (runtimePath: string, project: IBotProject) => Promise<void>;
+  build: (runtimePath: string, project: IBotProject, fullSettings?: DialogSetting, port?: string) => Promise<void>;
```

Comment:
Windows app services listen on a named pipe which is expressed as a string. Unix sockets are also strings.

Reply:
Doubtful, but if this code interacts with deployments at all it may be wise to keep that option open. If it's purely for local execution of bots, good chance you can limit it to numbers.